.github/workflows/ci.yml

```diff
@@ -0,0 +1,28 @@
+name: Java CI with Maven
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Get Java Version
+        run: |
+          Java_Version=$(mvn help:evaluate "-Dexpression=maven.compiler.release" -q -DforceStdout | sed -e 's/^1\./1.0./')
+          echo "Java_Version=$Java_Version" >> $GITHUB_ENV
```

Comment:
Is this in case the Java version of the project changes?

Reply:
If that is the case, it looks like a cool feature to have :)

Reply:
Yes, it takes the java version from pom.xml config so we always install the java version that maven are going to build the project with.